Simple utility script to clone Samudra data locally.

alxmrs opened the pull request:
Clones OM4 in about 600s using ~100MB chunks. Saves means and stds as NetCDF. Is packaged as a `uv` script (it self-describes dependencies).

This change is in service of setting up remote GPU runs and chunk tuning.

alxmrs committed:
Simple utility script to clone Samudra data locally.

Clones OM4 in about 600s using ~100MB chunks. Saves means and stds as NetCDF. Is packaged as a `uv` script (it self-describes dependencies).

This change is in service of setting up remote GPU runs and chunk tuning.

alxmrs committed:
Updated uv.lock.

suryadheeshjith reviewed (approved):
LGTM! I am curious as to why exactly we are saving means and stds as netcdf files

suryadheeshjith commented on scripts/clone_data.py:
```diff
@@ -0,0 +1,70 @@
+                DATA_ROOT + name, engine="zarr", chunks={"time": 700}
+            )
+            data = data.isel(time=time_slice)
+            data.chunk({"time": 1})
```
```suggestion
            data = data.chunk({"time": 1})
```

chunk() is not an in-place operation

suryadheeshjith replied:
Based on my tests with it, it seems to slow down saving when we rechunk with time=1

alxmrs replied:
That would make sense to me! Though, changing the chunks at rest is a significant data engineering change. I'd rather save stuff that like for the future.

jder commented:
@alxmrs what version of uv are you using? It seems like we keep getting this trivial change back and forth in our uv.lock and I think it must be mismatching of versions. I'm running uv 0.6.5.

jder commented:
@suryadheeshjith all the other scripts in this directory start with a `.` -- is that historical or is there some reason this one should too?

jder reviewed (approved):
LGTM after Surya's feedback.

suryadheeshjith commented:
No reason, just to indicate it is a bash script

alxmrs committed:
Update scripts/clone_data.py

Co-authored-by: Surya Dheeshjith <41594351+suryadheeshjith@users.noreply.github.com>

alxmrs commented:
I just committed surya's suggestion; going to test it out now (on my new dev machine! 😀 ).

alxmrs commented:
>  I am curious as to why exactly we are saving means and stds as netcdf files

I though that the train pipeline expects that these are nc files -- am I wrong about this?

suryadheeshjith commented:
~~It need not be but it can!~~ I am mistaken, it does need to be netcdf files. I am getting confused working with a million repositories. sorry.

alxmrs committed:
Improvements to clone script.

- More specific deps
- extracted out a common variable
- Moved chunks to a better place (not sure if it made a performance difference, but it looks cleaner).

alxmrs commented:
>  I'm running uv 0.6.5.

Yeah, I've been having some weird uv differences. I just did a fresh install of `uv` on my remote VM, and it is version: `uv 0.6.6`

alxmrs commented:
Just ran this script on a fresh VM. The specs have changed a bunch, so it's not quite a fair comparison, but this still downloads all the data in about 10m or 600s.
